File: CPP/program/Mahasiswa.cpp
/*
Saya Riska Nurohmah [2109103] mengerjakan Latihan Praktikum 4 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/

// Library
#include <iostream>
#include <string>

// using standard namespace
using namespace std;

// Membuat kelas Mahasiswa yang merupakan anak dari kelas Sivitas Akademik dan cucu dari kelas Human
class Mahasiswa : public SivitasAkademik
{
    // deklarasi atribut privat
private:
    string nim;
    string fakultas;

    // deklarasi method public
public:
    // konstruktor
    Mahasiswa()
    {
    }
    // method untuk mengset atribut nim
    void setNim(string nim)
    {
        this->nim = nim;
    }
    // method untuk mengset atribut fakultas
    void setFakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }
    // method untuk mengambil nilai atribut nim
    string getNim()
    {
        return this->nim;
    }
    // method untuk mengambil nilai atribut fakultas
    string getFakultas()
    {
        return this->fakultas;
    }
    // destruktor
    ~Mahasiswa()
    {
    }
};

File: Python/program/Mahasiswa.py
# Saya Riska Nurohmah [2109103] mengerjakan Latihan Praktikum 4 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

# import class atau library SivitasAkademik
from SivitasAkademik import SivitasAkademik

# Membuat kelas Mahasiswa yang merupakan anak dari kelas SivitasAkademik dan cucu dari kelas Human
# sehingga kelas Mahasiswa dapat mengakses atribut dan method dari kelas SivitasAkademik maupun HUman


class Mahasiswa(SivitasAkademik):
    # deklarasi atribut privat
    __nim = ""
    __fakultas = ""

    # konstruktor
    def __Init__(self, nim="", fakultas="", prodi=""):
        self.__nim = nim
        self.__fakultas = fakultas

    # method setter dan getter
    def setNim(self, nim):
        self.__nim = nim

    def setFakultas(self, fakultas):
        self.__fakultas = fakultas

    def getNim(self):
        return self.__nim

    def getFakultas(self):
        return self.__fakultas
